### PR TITLE
Frontend: Added Notifications + Adjusted Notification Endpoints

### PR DESCRIPTION
**Pull request by FurKay00**

### CODE REVIEW

File: src/main/web/package.json [hunk 1]
```diff
@@ -42,23 +44,24 @@
     "nspell": "^2.1.5",
     "nth-check": "^2.0.1",
     "quill": "^2.0.2",
-    "react": "^18.3.0",
+    "react": "^18.3.1",
```
Comment (maxschwinghammer) [inline src/main/web/package.json:47]: Change this back to version `^18.3.0`

File: src/main/web/package.json [hunk 2]
```diff
@@ -42,23 +44,24 @@
...
     "react-animated-heart": "^0.0.8",
     "react-auth-code-input": "^3.2.1",
     "react-badge": "^1.0.0",
     "react-big-calendar": "^1.12.2",
     "react-dom": "^18.3.1",
     "react-images-uploading": "^3.1.7",
-    "react-leaflet": "^4.2.1",
+    "react-leaflet": "^3.2.5",
```
Comment (maxschwinghammer) [inline src/main/web/package.json:54]: Change this back to version `4.2.1`

File: src/main/web/src/organisms/header/Notifications.css
Comment (maxschwinghammer) [inline src/main/web/src/organisms/header/Notifications.css:1]: Delete comments

File: src/main/web/src/organisms/header/Notifications.tsx [hunk 3]
```diff
@@ -1,45 +1,84 @@
...
 
 interface NotificationsProps {
   showNotifications: boolean;
+  notifications: NotificationModel[];
+  setNotifications: Dispatch<SetStateAction<NotificationModel[]>>;
 }
 
-interface NavigatorWithBadge extends Navigator {
-  setAppBadge?: (count: number) => Promise<void>;
+export interface NotificationModel {
```
Comment (maxschwinghammer) [inline src/main/web/src/organisms/header/Notifications.tsx:13]: Best practise: Extract this to a seperate file called `NotificationModel` in a folder `models`

File: src/main/web/src/organisms/header/Notifications.tsx [hunk 4]
```diff
@@ -1,45 +1,84 @@
...
+  notificationId: number;
+  groupId: number;
+  text: string;
+  link: string;
+  type: string;
 }
 
-export const Notifications: React.FC<NotificationsProps> = ({ showNotifications }) => {
-  const [notificationShown, setNotificationShown] = useState(false);
+interface DeleteNotificationModel {
```
Comment (maxschwinghammer) [inline src/main/web/src/organisms/header/Notifications.tsx:21]: Best practise: Extract this to a seperate file called `DeleteNotificationModel` in a folder `models`

File: src/main/web/src/organisms/login/Login.tsx [hunk 5]
```diff
@@ -66,7 +66,7 @@ const Login: React.FC<Props> = () => {
     const handleGoogleLogin = (credentialResponse: CredentialResponse): void => {
         googleLogin(JSON.stringify({ token: credentialResponse.credential })).then(
             (): void => {
-                navigate("/profile");
+                navigate("/");
```
Comment (maxschwinghammer) [inline src/main/web/src/organisms/login/Login.tsx:69]: Can then completely removed

**Review by maxschwinghammer — CHANGES_REQUESTED** [review]
See comments

**Review by denniskp — APPROVED** [review]

**Review by Nahlam4 — APPROVED** [review]

**Review by maxschwinghammer — APPROVED** [review]

**Pull request merged**